api/src/main/kotlin/io/spine/protodata/protobuf/FieldDescriptorExts.kt

```diff
@@ -84,45 +83,25 @@ import kotlin.reflect.KClass
  */
 public fun FieldDescriptor.name(): FieldName = fieldName { value = name }
 
-/**
- * Converts this field descriptor into a [Field] with options.
- *
- * @see buildField
- */
-public fun FieldDescriptor.toField(): Field {
-    val self = this
-    val field = buildField(self)
-    return field.copy {
-        // There are several similar expressions in this file, like
-        // the `option.addAll()` call below. Sadly, these duplicates
-        // could not be refactored into a common function because
-        // they have no common compile-time type.
-        option.addAll(options())
-    }
-}
-
 /**
  * Converts this field descriptor into a [Field].
- *
- * The resulting [Field] will not reflect the field options.
- *
- * @see toField
  */
-public fun buildField(desc: FieldDescriptor): Field =
+public fun FieldDescriptor.toField(): Field =
     field {
-        val messageType = desc.containingType
+        val messageType = containingType
         val declaredIn = messageType.name()
-        name = desc.name()
+        name = name()
         // New `FieldType` and `group_name` API.
-        type = desc.toFieldType()
+        type = toFieldType()
         declaringType = declaredIn
-        number = desc.number
-        orderOfDeclaration = desc.index
-        desc.realContainingOneof?.let {
+        number = number
```

Comment:
This `number = number` looks suspicious. Perhaps a reference to the context would make it clearer.